src/main/java/hudson/plugins/cobertura/CoberturaPublisher.java

```diff
@@ -43,64 +46,46 @@
 
 import org.apache.commons.beanutils.ConvertUtils;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 import org.kohsuke.stapler.StaplerRequest;
 
 /**
  * Cobertura {@link Publisher}.
  *
  * @author Stephen Connolly
  */
-public class CoberturaPublisher extends Recorder {
+public class CoberturaPublisher extends Recorder implements SimpleBuildStep {
+
+    private String coberturaReportFile;
 
-    private final String coberturaReportFile;
+    private boolean onlyStable;
 
-    private final boolean onlyStable;
+    private boolean failUnhealthy;
 
-    private final boolean failUnhealthy;
+    private boolean failUnstable;
 
-    private final boolean failUnstable;
+    private boolean autoUpdateHealth;
 
-    private final boolean autoUpdateHealth;
+    private boolean autoUpdateStability;
 
-    private final boolean autoUpdateStability;
+    private boolean zoomCoverageChart;
+
+    private int maxNumberOfBuilds = 0;
 
-    private final boolean zoomCoverageChart;
-    
-    private final int maxNumberOfBuilds;
-    
     private boolean failNoReports = true;
 
-    private CoverageTarget healthyTarget;
+    private CoverageTarget healthyTarget = new CoverageTarget();
 
-    private CoverageTarget unhealthyTarget;
+    private CoverageTarget unhealthyTarget = new CoverageTarget();
 
-    private CoverageTarget failingTarget;
+    private CoverageTarget failingTarget = new CoverageTarget();
 
     public static final CoberturaReportFilenameFilter COBERTURA_FILENAME_FILTER = new CoberturaReportFilenameFilter();
 
-    private final SourceEncoding sourceEncoding;
+    private SourceEncoding sourceEncoding = SourceEncoding.UTF_8;
 
-    /**
-     * @param coberturaReportFile the report directory
-     * @stapler-constructor
-     */
     @DataBoundConstructor
-    public CoberturaPublisher(String coberturaReportFile, boolean onlyStable, boolean failUnhealthy, boolean failUnstable, 
```

Comment:
When I did the switch, I didn't run into the backwards compatibility issues; it seemed like changing from a single DataBoundConstructor to multiple DataBoundSetter methods didn't break compatibility on the surface. We could restore the constructor, but it's now redundant, and only used in the very tests that were just re-added. However, I'm not an expert on this; opinions?